src/Views/Timeline.vala

```diff
@@ -114,13 +134,24 @@ public class Tootle.Views.Timeline : Views.Abstract {
         }
 
         var msg = new Soup.Message ("GET", get_url ());
+        if(accounts.currentInstance.version.ascii_casecmp ("3.0.0")>=0 && this.timeline=="direct"){
+            msg = new Soup.Message ("GET", get_url_status ());
+        }
```

Comment:
What do you think about decoupling this conditional logic to different function to avoid unneeded network request?

Reply:
It sounds nice, I'm going to update the code with a proposal.